File: src/main/java/net/obesephoenix/koalakraftcrystals/events/onItemDrop.java
package net.obesephoenix.koalakraftcrystals.events;

import net.obesephoenix.koalakraftcrystals.KoalaKraftCrystals;
import net.obesephoenix.koalakraftcrystals.crystals.GodModeHandler;
import net.obesephoenix.koalakraftcrystals.crystals.KKCrystalHandler;
import net.obesephoenix.koalakraftcrystals.util.KKMessage;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDropItemEvent;
import org.bukkit.event.player.PlayerDropItemEvent;
import org.bukkit.inventory.ItemStack;
import org.bukkit.scheduler.BukkitRunnable;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class onItemDrop implements Listener {

    private static final Map<ItemStack, Player> toDrop = new HashMap<>();

    @EventHandler
    public void onEvent(PlayerDropItemEvent e) {
        Player p = e.getPlayer();
        ItemStack i = e.getItemDrop().getItemStack();

        if (KKCrystalHandler.isCrystal(i)) {
            if (!toDrop.containsKey(i)) {
                e.setCancelled(true);
                List<String> messages = KKMessage.formatMultiline("no-drop");
                messages.forEach(p::sendMessage);
                toDrop.put(i, p);

                new BukkitRunnable() {
                    @Override
                    public void run() {
                        toDrop.remove(i, p);
                    }
                }.runTaskLater(KoalaKraftCrystals.instance, 100L);
            } else {
                toDrop.remove(i, p);
            }
        }
    }
}

File: src/main/java/net/obesephoenix/koalakraftcrystals/events/OnItemDrop.kt
package net.obesephoenix.koalakraftcrystals.events

import net.obesephoenix.koalakraftcrystals.crystals.KKCrystalHandler.isCrystal
import org.bukkit.event.player.PlayerDropItemEvent
import org.bukkit.entity.Player
import net.obesephoenix.koalakraftcrystals.util.KKMessage
import org.bukkit.scheduler.BukkitRunnable
import net.obesephoenix.koalakraftcrystals.KoalaKraftCrystals
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.inventory.ItemStack
import java.util.HashMap
import java.util.function.Consumer

class OnItemDrop : Listener {
    @EventHandler
    fun onEvent(e: PlayerDropItemEvent) {
        val p = e.player
        val i = e.itemDrop.itemStack
        if (isCrystal(i)) {
            if (!toDrop.containsKey(i)) {
                e.isCancelled = true
                val messages = KKMessage.formatMultiline("no-drop")
                messages.forEach(Consumer { s: String? -> p.sendMessage(s) })
                toDrop[i] = p
                object : BukkitRunnable() {
                    override fun run() {
                        toDrop.remove(i, p)
                    }
                }.runTaskLater(KoalaKraftCrystals.instance, 100L)
            } else {
                toDrop.remove(i, p)
            }
        }
    }

    companion object {
        private val toDrop: MutableMap<ItemStack, Player> = HashMap()
    }
}